### PR TITLE
[DSS] DDP-7921: Comparison rule API specification updated

### PR DESCRIPTION
**Pull request by Trimax**
1. Update API specification by adding COMPARISON_RULE type
2. The ComparisonRule model class updated by adding referenceQuestionStableId

### CODE REVIEW

**Review by ssettipalli — DISMISSED** [review]

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dao/ValidationDao.java [hunk 1]
```diff
@@ -253,6 +254,11 @@ private Rule getValidationRule(RuleDto dto, String message, String hint) {
                         .correctionHint(hint)
                         .allowSave(dto.isAllowSave())
                         .referenceQuestionId(comparisonRule.getReferenceQuestionId())
+                        .referenceQuestionStableId(TransactionWrapper.withTxn(handle ->
+                                handle.attach(QuestionDao.class).getJdbiQuestion()
+                                        .findQuestionDtoById(comparisonRule.getReferenceQuestionId())
+                                        .map(QuestionDto::getStableId)
+                                        .orElse(null)))
                         .comparisonType(comparisonRule.getType())
```
Comment (bskinner) [inline pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dao/ValidationDao.java]: What options do we have aside from triggering another database round-trip _just_ for the `stableId` here? My impression is all of the Rule DTOs are composed using the results from the `select_all_question_rule_dtos` statement. Can the `select_all_question_rule_dtos` query be extended to include the `stableId` in the initial round trip?

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 2]
```diff
@@ -2270,6 +2271,28 @@ Validation.UniqueValue:
...
+    - type: object
+      description: compares the value of the given answer with the answer to the referenced question
+      properties:
+        rule:
+          type: string
+          default: COMPARISON
+          enum:
+            - COMPARISON
+        reference_question_stable_id:
+          type: string
```
Comment (bskinner) [inline pepper-apis/docs/specification/src/components/schemas.yml:2289]: Let's add a `format: guid` property to this field

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 3]
```diff
@@ -2270,6 +2271,28 @@ Validation.UniqueValue:
           default: UNIQUE_VALUE
           enum:
             - UNIQUE_VALUE
+Validation.Comparison:
+  allOf:
+    - $ref: '#/Validation'
+    - type: object
+      description: compares the value of the given answer with the answer to the referenced question
+      properties:
```
Comment (bskinner) [inline pepper-apis/docs/specification/src/components/schemas.yml:2282]: Which of the below properties are `required`?

**Review by bskinner — CHANGES_REQUESTED** [review]

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dao/ValidationDao.java (hunk 1 above)
Comment (Trimax) [reply pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dao/ValidationDao.java]: Actually, this is a good idea. Let me update DTO & SQL

**Review by bskinner — APPROVED** [review]
Thanks for the changes! I'm good with merging this.

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**